spec/features/post_index_spec.rb

```diff
@@ -1,45 +1,32 @@
 require 'rails_helper'
-RSpec.describe 'Post Index Page', type: :feature do
-  describe 'Viewing Post Index page' do
-    before(:each) do
-      @user1 = User.create(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.',
-                           posts_counter: 0)
-      @first_post = Post.create(author: @user1, title: 'post1', text: 'This is my first post', comments_counter: 0,
-                                likes_counter: 0)
-      @comment1 = Comment.create(post: @first_post, author: @user1, text: 'Hi Tom!, Nice comment')
-      visit user_posts_path(@user1)
-    end
 
-    it 'should show user name' do
-      expect(page).to have_content(@user1.name)
-    end
-
-    it 'should show user photo' do
-      expect(page).to have_selector('img[alt="Tom"]')
-    end
-
-    it 'should show posts_counter of clicked user' do
-      expect(page).to have_content(@user1.posts_counter)
-    end
-
-    it 'should show post title' do
-      expect(page).to have_content(@first_post.title)
-    end
-
-    it 'should show post text' do
-      expect(page).to have_content(@first_post.text)
-    end
-
-    it 'should show comment text' do
-      expect(page).to have_content(@comment1.text)
-    end
+RSpec.feature 'Post Index', type: :feature do
+  let(:user) { User.create(name: 'Tom', photo: 'https://www.kasandbox.org/programming-images/avatars/leaf-blue.png', bio: 'He is a good programmar') }
+  let!(:post) { Post.create(author: user, title: "first post's title", text: 'first text') }
+  let!(:comment1) { Comment.create(author: user, post:, text: 'first comment') }
+  let!(:comment2) { Comment.create(author: user, post:, text: 'second comment') }
+  let!(:comment3) { Comment.create(author: user, post:, text: 'third comment') }
+  let!(:like1) { Like.create(author: user, post:) }
+
+  scenario "see user's profile picture, username, number of posts and interactions" do
+    visit user_posts_path(user)
+    expect(page).to have_content('Tom')
+    expect(page).to have_css("img[alt='Tom']", count: 1)
+    expect(page).to have_content('1 post')
+    expect(page).to have_content('Comments: 3')
+    expect(page).to have_content('Likes: 1')
+  end
 
```

Comment:
- [ ] As requested by the previous reviewer add the below test for the `posts#index` page, You still missed adding that. Please make sure to add that.
![image](https://github.com/VaheStepanyan100/blog-app/assets/83803414/ca62ae83-127a-45a8-9f20-a707945ea103)